### PR TITLE
Make PluginContainer injectable

### PR DESCRIPTION
**Pull request by alexstaeding**
This PR makes `PluginContainer` injectable in a plugin.

### CODE REVIEW

**Comment by Crypnotic** [comment]
I really only see this being practical if we could adapt this so that plugins could inject instances of other plugins that they depend on. That would allow inter-plugin loading more stream-lined. Other than that, this is information that can be gathered by other means for the few people who actually need it

**Comment by alexstaeding** [comment]
The latest commit 0637f6229dbed0332d8bfe49afa48c4a8d856c95 lets plugins inject the `PluginContainer` from any other loaded plugin by writing:
```java
@Inject
@Named("<pluginid>")
PluginContainer pluginContainer;
```
And if they want to inject their own `PluginContainer`:
```java
@Inject
PluginContainer pluginContainer;
```

File: proxy/src/main/java/com/velocitypowered/proxy/plugin/loader/PluginLoader.java [hunk 1]
```diff
@@ -11,5 +13,30 @@
...
-  PluginContainer createPlugin(PluginDescription plugin) throws Exception;
+  /**
+   * Creates a {@link Module} for the provided {@link PluginContainer}
+   * and verifies that the container's {@link PluginDescription} is correct.
+   *
+   * <p>Does not create an instance of the plugin.</p>
+   *
+   * @param container the plugin container
+   * @return the module containing bindings specific to this plugin
+   * @throws Exception If anything went wrong
```
Comment (astei) [inline proxy/src/main/java/com/velocitypowered/proxy/plugin/loader/PluginLoader.java]: Can we be more specific here? This bit of JavaDoc adds no information.

File: proxy/src/main/java/com/velocitypowered/proxy/plugin/VelocityPluginManager.java [hunk 2]
```diff
@@ -92,19 +100,44 @@ public void loadPlugins(Path directory) throws IOException {
...
+    // Make a global Guice module that with common bindings for every plugin
+    AbstractModule commonModule = new AbstractModule() {
+      @Override
+      protected void configure() {
+        bind(ProxyServer.class).toInstance(server);
+        bind(PluginManager.class).toInstance(server.getPluginManager());
+        bind(EventManager.class).toInstance(server.getEventManager());
+        bind(CommandManager.class).toInstance(server.getCommandManager());
+        for (PluginContainer container : pluginContainers.keySet()) {
+          bind(PluginContainer.class)
```
Comment (astei) [inline proxy/src/main/java/com/velocitypowered/proxy/plugin/VelocityPluginManager.java]: If you try to inject a `PluginContainer` for some other plugin but that plugin doesn't exist, Guice will refuse to initialize the object for the plugin. Given that Velocity also supports optional dependencies, this isn't some strange thing that will never happen - it is a possibility.

Comment (alexstaeding) [reply proxy/src/main/java/com/velocitypowered/proxy/plugin/VelocityPluginManager.java]: Guice supports optional dependencies with
```java
@Inject(optional = true)
```

**Pull request merged**